src/main/java/net/starlark/java/eval/MethodDescriptor.java

```diff
@@ -100,6 +102,18 @@ private MethodDescriptor(
     } else {
       howToHandleReturn = HowToHandleReturn.FROM_JAVA;
     }
+
+    if (extraKeywords || extraPositionals || useStarlarkSemantics || useStarlarkThread) {
+      this.canReusePositionalWithoutChecks = false;
+    } else if (!Arrays.stream(parameters).allMatch(MethodDescriptor::paramCanBeUsedAsPositionalWithoutChecks)) {
+      this.canReusePositionalWithoutChecks = false;
+    } else {
+      this.canReusePositionalWithoutChecks = true;
+    }
+  }
+
+  private static boolean paramCanBeUsedAsPositionalWithoutChecks(ParamDescriptor param) {
+    return param.isPositional() && param.disabledByFlag() == null && param.getAllowedClasses().contains(Object.class);
```

Comment:
Actually, this check seems to be too conservative: we can still apply the optimization if non-positional parameters exist, as long as they are not used during the call.